extensions/data-workspace/src/dialogs/projectDashboard.ts

```diff
@@ -0,0 +1,93 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as constants from '../common/constants';
+import { IProjectProvider, WorkspaceTreeItem } from 'dataworkspace';
+import { IWorkspaceService } from '../common/interfaces';
+export class ProjectDashboard {
+
+	private dashboard: azdata.window.ModelViewDashboard | undefined;
+	private modelView: azdata.ModelView | undefined;
+	private projectProvider: IProjectProvider | undefined;
+	private overviewTab: azdata.DashboardTab | undefined;
+
+	constructor(private workspaceService: IWorkspaceService, private treeItem: WorkspaceTreeItem) {
+	}
+
+	public async showDashboard(): Promise<void> {
+		const project = this.treeItem.element.project;
+		const projectTypeId = project.projectFilePath;
+		this.projectProvider = await this.workspaceService.getProjectProvider(vscode.Uri.file(projectTypeId));
+		if (!this.projectProvider) {
+			throw new Error(constants.ProviderNotFoundForProjectTypeError(projectTypeId));
+		}
+
+		await this.createDashboard(project.projectFileName);
+		await this.dashboard!.open();
+	}
+
+	private async createDashboard(title: string): Promise<void> {
+		this.dashboard = azdata.window.createModelViewDashboard(title, undefined, { alwaysShowTabs: false });
+		this.dashboard.registerTabs(async (modelView: azdata.ModelView) => {
+			this.modelView = modelView;
+
+			this.overviewTab = {
+				title: '',
+				id: 'overview-tab',
+				content: this.createContainer(title),
+				toolbar: await this.createToolbarContainer()
+			};
+			return [
+				this.overviewTab
+			];
+		});
+	}
+
+	private async createToolbarContainer(): Promise<azdata.ToolbarContainer> {
+		const projectActions = await this.workspaceService.getProjectToolbarActions(this.projectProvider!);
+
+		// Add actions as buttons
```

Comment:
will other project actions be allowed to be something besides buttons eventually? Like dropdowns?

Reply:
Created this issue to track it: https://github.com/microsoft/azuredatastudio/issues/14190